x/evm/keeper/msg_server.go

```diff
@@ -1015,13 +1015,15 @@ func (s msgServer) CreateDeployToken(c context.Context, req *types.CreateDeployT
 
 	keeper := s.ForChain(chain.Name)
 
-	originChain, found := s.nexus.GetChain(ctx, req.Asset.Chain)
-	if !found {
-		return nil, fmt.Errorf("%s is not a registered chain", req.Asset.Chain)
-	}
+	if req.Address.IsZeroAddress() {
+		originChain, found := s.nexus.GetChain(ctx, req.Asset.Chain)
+		if !found {
+			return nil, fmt.Errorf("%s is not a registered chain", req.Asset.Chain)
+		}
 
-	if !s.nexus.IsAssetRegistered(ctx, originChain, req.Asset.Name) {
-		return nil, fmt.Errorf("asset %s is not registered on the origin chain %s", originChain.NativeAsset, originChain.Name)
+		if !s.nexus.IsAssetRegistered(ctx, originChain, req.Asset.Name) {
+			return nil, fmt.Errorf("asset %s is not registered on the origin chain %s", originChain.NativeAsset, originChain.Name)
+		}
```

Comment:
why dont' we need to check if the asset is registered for external tokens?

Reply:
Cuz when deploying an external token, this is the process of having it registered.